chess_ng/board.py

```diff
@@ -28,7 +29,7 @@ def __init__(self, pieces: Optional[List[Piece]] = None, size: int = 8):
             for pos in itertools.product(range(size), repeat=2)
         }
         self.size = size
-        self.move_history: List[Tuple[Piece, Tuple[int, int]]] = []
+        self.move_history: List[Tuple[Piece, Tuple[int, int], bool]] = []
```

Comment:
I feel like a tuple of 3 calls for a little dataclass to hold the attributes instead, to avoid indexing errors, but for now this is probably ok 👍